finality-aleph/src/sync/mock/backend.rs

```diff
@@ -254,7 +233,37 @@ impl Finalizer<MockJustification> for Backend {
 
 impl BlockImport<MockBlock> for Backend {
     fn import_block(&mut self, block: MockBlock) {
-        self.import(block.header);
+        if !block.verify() {
+            return;
+        }
+
+        let mut storage = self.inner.lock();
+
+        let parent_id = match block.parent_id() {
+            Some(id) => id,
+            None => return,
+        };
+
+        if storage.blockchain.contains_key(&block.id())
```

Comment:
we could add a check that the block being imported is not below the last finalized

Reply:
check lgtm